site/docs/features/index.md

```diff
@@ -1,107 +1,327 @@
+
+Since data lakes contain the history of all that data, it is possible to
+perform queries against the state of the data as it was at a point in time
+in the past.
+
+The amount of data held in data lakes is rather huge (GBs, TBs, PBs), and so
+is the number of tables and data files (100s of thousands, millions).
+
+Managing that amount of data and data files while keeping track of schema
+changes, for example adding or removing a column, changing a column's type,
+renaming a column in a table ("PDS", physical datasets) and views ("VDS",
+virtual datasets), is one of the things that Nessie tackles.
+
+Data in a data lake is usually consumed and written using tools like
+Apache [Hive](https://hive.apache.org)[^2] or Apache
+[Spark](https://spark.apache.org)[^2]. Your existing jobs can easily integrate
+Nessie without any production code changes, it's a simple configuration change.
+
+## git 101
+
+> *"Git is a free and open source distributed version control system designed to 
+handle everything from small to very large projects with speed and efficiency"*
+(cite from [git-scm.com](https://git-scm.com/))
+
+git maintains the history or all changes of a software project from the very
+first *commit* until the current state.
+
+git is used by humans, i.e. developers.
+
+Many of the concepts of git for source code are implemented by Nessie for all
+the data your data lake. It would be rather confusing to explain all git
+concepts here and then outline the differences in the next chapter.
+
+## Working with data in Nessie
+
+Each individual state in Nessie is defined by a *Nessie commit*.
+Each *commit* in Nessie, except the very first one, has references to its
+predecessors, the previous versions of the data.
+
+Each *Nessie commit* also "knows" about the data files in your data lake, which
+represent the state of all data in all tables.
+
+The following example illustrates that our *current commit* adds a 3rd data file.
+The other two data files 1+2 have been added by *previous commit*.
 ```
-$ create t1
-...
-$ insert 2 records into t1
-...
-$ create t2
-...
-$ insert 2 records into t2
-...
+ +-------------------+       +-------------------------+
+ |  previous commit  | --<-- |     current commit      |
+ +-------------------+       +-------------------------+
+     |         |                 |        |        |
+   (add)     (add)               |        |      (add)
+     |         |                 |        |        |
+  +------+  +------+          +------+ +------+ +------+
+  | data |  | data |          | data | | data | | data |
+  | file |  | file |          | file | | file | | file |
+  | #1   |  | #2   |          | #1   | | #2   | | #3   |
+  |     _|  |     _|          |     _| |     _| |     _|
+  |  __/    |  __/            |  __/   |  __/   |  __/  
+  |_/       |_/               |_/      |_/      |_/     
 ```
+In "relational SQL" you can think of the following sequence of SQL statements:
```

Comment:
Maybe too technical but pointing out the major difference between normal transactions and Nessie transactions is that they can be distributed: eg 1 spark job could modify `table_one` and another spark job could modify `other_table` in the same _transaction_. Important distinction but perhaps for a more technical audience

Reply:
Doesn't that actually depend on how you actually run your Spark job with Nessie?
IIUC there are many ways how users can use Nessie "transactions" --> Nessie commits
But yea, listing (at least some of) these options is probably helpful (and hopefully not confusing).

Reply:
yea, 1 commit is always going to be on 1 spark instance. But if several spark instances commit to the same branch you can achieve a distributed transaction (sort of). This might be an intermediate concept though but it is an important differentiator compared to normal sql databases

Reply:
Added some more words (hopefully not too complicated).